core/src/main/scala/cromwell/core/logging/WorkflowLogger.scala

```diff
@@ -0,0 +1,101 @@
+package cromwell.core.logging
+
+import java.nio.file.{Path, Paths}
+
+import akka.actor.{Actor, ActorLogging}
+import akka.event.LoggingAdapter
+import better.files._
+import ch.qos.logback.classic.encoder.PatternLayoutEncoder
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.{Level, LoggerContext}
+import ch.qos.logback.core.FileAppender
+import com.typesafe.config.ConfigFactory
+import cromwell.core.WorkflowId
+import lenthall.config.ScalaConfig._
+import org.slf4j.helpers.NOPLogger
+import org.slf4j.{Logger, LoggerFactory}
+
+trait WorkflowLogging extends ActorLogging { this: Actor =>
+  def workflowId: WorkflowId
+
+  lazy val workflowLogger = new WorkflowLogger(self.path.name, workflowId, Option(log))
+}
+
+object WorkflowLogger {
+  def makeFileLogger(path: Path, level: Level): Logger = {
+    val ctx = LoggerFactory.getILoggerFactory.asInstanceOf[LoggerContext]
+    val name = path.getFileName.toString
+
+    Option(ctx.exists(name)) match {
```

Comment:
It's good that there's a `ConcurrentHashMap` to keep the logger factory's internals from getting out of whack, but it still seems possible that multiple threads could create different `FileAppender`s for the same underlying log file and hand them back to their callers. (i.e. the get/set for the logger by name is not atomic). I can't remember ever seeing any issues like this in the Olde Worlde so maybe I'm just being paranoid.